src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDocumentSynchronizationEndpoint.cs

```diff
@@ -166,12 +117,15 @@ internal SourceText ApplyContentChanges(IEnumerable<TextDocumentContentChangeEve
                     throw new ArgumentNullException(nameof(change.Range), "Range of change should not be null.");
                 }
 
-                var linePosition = new LinePosition(change.Range.Start.Line, change.Range.Start.Character);
-                var position = sourceText.Lines.GetPosition(linePosition);
-                var textSpan = new TextSpan(position, change.RangeLength);
+                var startLinePosition = new LinePosition(change.Range.Start.Line, change.Range.Start.Character);
+                var startPosition = sourceText.Lines.GetPosition(startLinePosition);
+                var endLinePosition = new LinePosition(change.Range.End.Line, change.Range.End.Character);
+                var endPosition = sourceText.Lines.GetPosition(endLinePosition);
+
+                var textSpan = new TextSpan(startPosition, change.RangeLength ?? endPosition - startPosition);
```

Comment:
This gets a bit verbose instead of just using `AsTextSpan` because we need to use the RangeLength if it's not null.